**Assignment 05**
Note: You will be using Pandas for this assignment.

Explore College Scorecard website - https://collegescorecard.ed.gov/

Explore College Scorecard Data web page - https://collegescorecard.ed.gov/data/

Download and read the documentation - https://collegescorecard.ed.gov/assets/FullDataDocumentation.pdf

Download the dataset - https://ed-public-download.app.cloud.gov/downloads/CollegeScorecard_Raw_Data.zip (265MB Zip File)

Unzip the file into a folder in your personal computer (there are many files, one for each school year)

Upload the 2017-2018 school year file "MERGED2017_18_PP.csv" to Colab and create a Jupyter notebook to explore the dataset:

how many colleges or rows?

how many variables or columns?

Look at the first 5 colleges

look at the last 5 colleges

Look at a random sample of 5 colleges

Since there are too many variables to explore, many of which are not interesting. Create a smaller data frame with the following 9 variables:
UNITID
INSTNM
CITY
STABBR
ZIP
ADM_RATE
UGDS
TUITIONFEE_IN
MN_EARN_WNE_P10

Download and read the data dictionary file - https://collegescorecard.ed.gov/assets/CollegeScorecardDataDictionary.xlsx

Find out the definition of the above 9 variables and document the definitions in the Jupyter Notebook in a Markdown cell so that the readers of your notebook will know the meaning of these variables.

Explore the smaller data frame

Look at the first 10 colleges

look at the last 10 colleges

Look at a random sample of 10 colleges

Find and display UMBC

Find Univ. of Maryland College Park
Find and display Loyola College Maryland
Find and display Johns Hopskins
Calculate the min, max, and average in-state tution of the above four institutions.
Filter the smaller dataset further to only keep colleges from Maryland, Virgina, Washington DC, Delaware, and Pennsylvinia. Save the filtered dataset to a new file named "Five_States_Colleges.csv" for future analysis. Upload this file to your GitHub along with the notebook.

Note:

Make sure your notebook file is well organized using Markdown and code comments.
Make sure you use Google Colab

**Importing necessary libraries**

In [108]:
import pandas as pd

**Reading Merged 2017-2018 csv data as Dataframe using pandas**

In [109]:
df = pd.read_csv('MERGED2017_18_PP.csv')

/usr/local/lib/python3.6/dist-packages/IPython/core/interactiveshell.py:2718: DtypeWarning: Columns (1729) have mixed types.Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


**Using shape, able to identify the size of the data frame which gives in the format of (rows,columns)**

In [110]:
df.shape

(7112, 1986)

**There are 7112 rows and 1986 columns**

**Identifying the data types of all the columns in the dataframe**

In [111]:
df.dtypes

UNITID                        int64
OPEID                         int64
OPEID6                        int64
INSTNM                       object
CITY                         object
                             ...   
SCUGFFN                     float64
POOLYRS_FTFTAIDPCT          float64
FTFTPCTPELL_POOLED_SUPP     float64
FTFTPCTFLOAN_POOLED_SUPP    float64
SCUGFFN_POOLED              float64
Length: 1986, dtype: object

**Using head function, identifying the first 5 rows of the data frame**

In [112]:
df.head(5) 

UNITID    OPEID  ...  FTFTPCTFLOAN_POOLED_SUPP SCUGFFN_POOLED
0  100654   100200  ...                       NaN            NaN
1  100663   105200  ...                       NaN            NaN
2  100690  2503400  ...                       NaN            NaN
3  100706   105500  ...                       NaN            NaN
4  100724   100500  ...                       NaN            NaN

[5 rows x 1986 columns]

**Using head function, identifying the last 5 rows of the data frame**

In [113]:
df.tail(5)

UNITID    OPEID  ...  FTFTPCTFLOAN_POOLED_SUPP SCUGFFN_POOLED
7107  48312404   108163  ...                       NaN            NaN
7108  48387801  4223701  ...                       NaN            NaN
7109  48400201  4228101  ...                       NaN            NaN
7110  48778201  4221501  ...                       NaN            NaN
7111  49005401  4182601  ...                       NaN            NaN

[5 rows x 1986 columns]

**Using sample and specifying n = 5 ,Looking at a random sample of 5 colleges**

In [114]:
df.sample(n=5, random_state=0)

UNITID    OPEID  ...  FTFTPCTFLOAN_POOLED_SUPP SCUGFFN_POOLED
6145  481368  4214500  ...                       NaN            NaN
1143  152497   625700  ...                       NaN            NaN
2087  184481   710700  ...                       NaN            NaN
840   140012  2288400  ...                       NaN            NaN
4784  436438  2179900  ...                       NaN            NaN

[5 rows x 1986 columns]

- **UNITID : Unit ID for institution**

- **INSTNM : Institution name**

- **CITY : City**

- **STABBR : State postcode**

- **ZIP : Zip postcode**

- **ADM_RATE : Admission rate**

- **UGDS : Enrollment of undergraduate certificate/degree-seeking students**

- **TUITIONFEE_IN : In-state tuition and fees**

- **MN_EARN_WNE_P10 : Mean earnings of students working and not enrolled 10 years after entry**

In [115]:
df_2 = df[['UNITID','INSTNM','CITY','STABBR','ZIP','ADM_RATE','UGDS','TUITIONFEE_IN','MN_EARN_WNE_P10']].copy()

**Looking at the first 10 colleges**


In [116]:
df_2.head(10)

UNITID                               INSTNM  ... TUITIONFEE_IN MN_EARN_WNE_P10
0  100654             Alabama A & M University  ...        9857.0             NaN
1  100663  University of Alabama at Birmingham  ...        8328.0             NaN
2  100690                   Amridge University  ...        6900.0             NaN
3  100706  University of Alabama in Huntsville  ...       10280.0             NaN
4  100724             Alabama State University  ...       11068.0             NaN
5  100751            The University of Alabama  ...       10780.0             NaN
6  100760    Central Alabama Community College  ...        4440.0             NaN
7  100812              Athens State University  ...           NaN             NaN
8  100830      Auburn University at Montgomery  ...        8020.0             NaN
9  100858                    Auburn University  ...       10968.0             NaN

[10 rows x 9 columns]

**looking at the last 10 colleges**

In [117]:
df_2.tail(10)

UNITID  ... MN_EARN_WNE_P10
7102  48154401  ...             NaN
7103  48305401  ...             NaN
7104  48312401  ...             NaN
7105  48312402  ...             NaN
7106  48312403  ...             NaN
7107  48312404  ...             NaN
7108  48387801  ...             NaN
7109  48400201  ...             NaN
7110  48778201  ...             NaN
7111  49005401  ...             NaN

[10 rows x 9 columns]

**Looking at a random sample of 10 colleges**

In [118]:
df_2.sample(n=10, random_state=0)

UNITID  ... MN_EARN_WNE_P10
6145    481368  ...             NaN
1143    152497  ...             NaN
2087    184481  ...             NaN
840     140012  ...             NaN
4784    436438  ...             NaN
4156    367024  ...             NaN
4424    407009  ...             NaN
378     118684  ...             NaN
5492    455275  ...             NaN
7019  44376603  ...             NaN

[10 rows x 9 columns]

**Finding and displaying UMBC**

In [119]:
df_3 = df_2.loc[df_2['INSTNM'] == 'University of Maryland-Baltimore County']

**Finding Univ. of Maryland College Park**

In [120]:
df_4 = df_2.loc[df_2['INSTNM'] == 'University of Maryland-College Park']

**Finding and displaying Loyola College Maryland**


In [121]:
df_5 = df_2.loc[df_2['INSTNM'] == 'Loyola University Maryland']

**Finding and displaying Johns Hopskins**


In [122]:
df_6 = df_2.loc[df_2['INSTNM'] == 'Johns Hopkins University']

**Calculate the min, max, and average in-state tution of the above four institutions.**

In [123]:
list_four_colleges = []

In [124]:
df_1_fee = df_2.loc[df_2['INSTNM'] == 'University of Maryland-Baltimore County', 'TUITIONFEE_IN'].iloc[0]
list_four_colleges.append(df_1_fee)

In [125]:
df_2_fee = df_2.loc[df_2['INSTNM'] == 'University of Maryland-College Park', 'TUITIONFEE_IN'].iloc[0]
list_four_colleges.append(df_2_fee)

In [126]:
df_3_fee = df_2.loc[df_2['INSTNM'] == 'Loyola University Maryland', 'TUITIONFEE_IN'].iloc[0]
list_four_colleges.append(df_3_fee)

In [127]:
df_4_fee = df_2.loc[df_2['INSTNM'] == 'Johns Hopkins University', 'TUITIONFEE_IN'].iloc[0]
list_four_colleges.append(df_4_fee)

In [128]:
list_four_colleges

[11518.0, 10399.0, 47560.0, 52170.0]

**Min Value in Four Colleges**

In [129]:
min(list_four_colleges)

10399.0

**Max Value in Four Colleges**

In [130]:
max(list_four_colleges)

52170.0

**Mean Value of Four Colleges in-state tution fees**

In [131]:
sum = 0
for i in list_four_colleges:
  sum+= i
print(sum)

121647.0


In [132]:
mean = sum / len(list_four_colleges)
print("The mean in-state tution fee value for mean", mean)

The mean in-state tution fee value for mean 30411.75


** Filter the smaller dataset further to only keep colleges from Maryland, Virgina, Washington DC, Delaware, and Pennsylvinia.**

In [133]:
df_5_collges = df_2.loc[df_2['STABBR'].isin(['MD','VA','DC','DE','PA'])]
df_5_collges

UNITID  ... MN_EARN_WNE_P10
634     130837  ...             NaN
635     130873  ...             NaN
636     130907  ...             NaN
637     130934  ...             NaN
638     130943  ...             NaN
...        ...  ...             ...
7063  45794301  ...             NaN
7084  47503303  ...             NaN
7089  47503308  ...             NaN
7094  47503313  ...             NaN
7103  48305401  ...             NaN

[690 rows x 9 columns]

**Save the filtered dataset to a new file named "Five_States_Colleges.csv" for future analysis. Upload this file to your GitHub along with the notebook.**

In [134]:
df_5_collges.to_csv(r'Five_States_Colleges.csv', index = False)